stepman/util.go

```diff
@@ -213,15 +270,102 @@ func ReadStepSpec() (models.StepCollectionModel, error) {
 	return stepCollection, err
 }
 
-// DownloadStep ...
-func DownloadStep(step models.StepModel) error {
-	gitSource := step.Source["git"]
-	pth := GetStepPath(step)
+// DownloadAndUnZIP ...
+func DownloadAndUnZIP(url, pth string) error {
+	filePath := os.TempDir() + "step.zip"
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			log.Fatal("Failed to close file:", err)
+		}
+		if err := os.Remove(filePath); err != nil {
+			log.Fatal("Failed to remove file:", err)
+		}
+	}()
+
+	response, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := response.Body.Close(); err != nil {
+			log.Fatal("Failed to close response body:", err)
+		}
+	}()
+
+	if response.StatusCode == http.StatusOK {
```

Comment:
do this in reverse: `if response.StatusCode != http.StatusOK` and return if error